### PR TITLE
Fix #315779: Disable auto-size of vertical frame when dragging the height handle

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: https://musescore.org/en/node/315779

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
There a few mtests that fail, seems those insist on 10 being the height of vertical frames. I don't quite get why and how thios is related to disabling the Autimatic site on dragging that size handle?

File: libmscore/box.cpp [hunk 1]
```diff
@@ -790,11 +790,24 @@ void VBox::layout()
                   contentHeight = minHeight();
 
             setHeight(contentHeight);
+            setBoxHeight(Spatium(contentHeight / spatium()));
```
Comment (Jojo-Schmitz) [inline libmscore/box.cpp:793]: Is this here the cause for the mtest fails?

Comment (jeetee) [reply libmscore/box.cpp:793]: Seems likely, as those now all get that default size set explicitly due to auto-size instead of keeping the old size value.

Comment (cbjeukendrup) [reply libmscore/box.cpp:793]: I think I have found a way of doing it without this line. I will try that soon.

**Comment by MarcSabatella** [comment]
That certainly looks a lot safer now.  Just thinking it through - what is the expected behavior if you turn autosize off for a box that has been expanded already buy this - should the box snap to the height given in the Inspector? And then let's say you add more content to the box, and now it's not big enough - should turning it on automatically snap the box to the required size?  Currently, both happen, and I think that's probably as expected.  It looks to me like this change won;t affect that, which is good, but also works as I expect with respect to dragging, where starting a drag on an autosize-expanded box disables autosize but does *not* cause the box instantly shrink.  In that sense it's very much like how Alt+drag of an autoplaced element disables autoplace but doesn't cause the position to suddenly reset.

So on paper all is good as far as I am concerned.

Once thing that has bugged me about the autosize design all along is that magic number 10 sp that basically serves as a minimum box height with autosize enabled.  I'd have preferred this to have been a style setting, assuming we could work out the compatibility implications.  FWIW, it's technically possible to implement that without much of a UI change, just add the style setting and mark the height property as styled in the boxStyle definition in box.cpp.  This should automatically result in a "set as style" button being added to the Inspector.  But, I'm not really recommending that right now - something to think about though.

File: libmscore/box.h [hunk 2]
```diff
@@ -153,6 +153,8 @@ class VBox : public Box {
       QVariant getProperty(Pid propertyId) const override;
       void layout() override;
 
+      virtual void startEditDrag(EditData&) override;
```
Comment (vpereverzev) [inline libmscore/box.h:156]: just curious, is there any reason to mark it virtual here?

**Comment by cbjeukendrup** [comment]
No, not really. It was just copied from `Box`, line 54. Should I remove it?

Op 3 feb. 2021, om 18:25 heeft pereverzev+v <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:



@vpereverzev commented on this pull request.

________________________________

In libmscore/box.h<https://github.com/musescore/MuseScore/pull/7362#discussion_r569601030>:

> @@ -153,6 +153,8 @@ class VBox : public Box {
       QVariant getProperty(Pid propertyId) const override;
       void layout() override;

+      virtual void startEditDrag(EditData&) override;


just curious, is there any reason to mark it virtual here?

File: libmscore/box.h (hunk 2 above)
Comment (cbjeukendrup) [reply libmscore/box.h:156]: No, not really. It was just copied from `Box`, line 54. Should I remove it?

Comment (vpereverzev) [reply libmscore/box.h:156]: Yeah, it's better to remove it, because it'll ported in master soon

Comment (cbjeukendrup) [reply libmscore/box.h]: Done!

Comment (vpereverzev) [reply libmscore/box.h]: cool, thanks

**Comment by igorkorsukov** [comment]
@cbjeukendrup Could you please port the changes to the master

**Comment by cbjeukendrup** [comment]
@igorkorsukov Yep, coming soon!

**Pull request merged**